server.js

```diff
@@ -0,0 +1,41 @@
+// server.js
+const { createServer } = require('http')
+const { parse } = require('url')
+const next = require('next')
+
+const dev = process.env.NODE_ENV !== 'production'
+const app = next({ dev })
+const handle = app.getRequestHandler()
+
+// this is the logger for the server
+var logger = require('pino-http')()
+
+app.prepare().then(() => {
+  createServer((req, res) => {
+    // Be sure to pass `true` as the second argument to `url.parse`.
+    // This tells it to parse the query portion of the URL.
+    const parsedUrl = parse(req.url, true)
+    const { pathname, query } = parsedUrl
+    logger(req, res)
+
+    if (pathname === '/log') {
+        let data = null
+        req.on('data', chunk => {
+            data = chunk
+        })
+        req.on('end', () => {
+            const { msg, level = 'info' } = JSON.parse(data)
+            req.log[level](msg)
+        })
+        res.statusCode = 200
+        res.setHeader('Content-Type', 'text/plain')
+        res.end('logged on server')
```

Comment:
please send the response only after we have recevied the body.

Reply:
ok I've made the change, have a look and let me know.